client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/use-validation-message.ts

```diff
@@ -88,6 +82,12 @@ export function useValidationMessage( domain: string, auth: string, hasDuplicate
 			loading: false,
 			message: __( 'This domain is unlocked but the authentication code seems incorrect.' ),
 		};
+	} else if ( availabilityNotice?.message ) {
+		return {
+			valid: false,
+			loading: false,
+			message: availabilityNotice?.message,
+		};
```

Comment:
This condition has to be here, otherwise we wouldn't show the correct message for an unlocked domain with incorrect auth code